src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -1,18 +1,33 @@
 (ns status-im.contexts.wallet.swap.events
   (:require [re-frame.core :as rf]
             [status-im.constants :as constants]
+            [status-im.contexts.communities.utils :as utils]
             [status-im.contexts.wallet.sheets.network-selection.view :as network-selection]
             [utils.number]))
 
 (rf/reg-event-fx :wallet.swap/start
- (fn [{:keys [_db]}]
+ (fn [{:keys [_]}]
    {:fx [[:dispatch [:open-modal :screen/wallet.swap-select-asset-to-pay]]]}))
 
+(rf/reg-event-fx :wallet.swap/start-with-token
+ (fn [{:keys [db]} [token]]
+   (let [current-address (get-in db [:wallet :current-viewing-account-address])
+         address         (:address (first (utils/sorted-operable-non-watch-only-accounts db)))
+         token-networks  (:networks token)]
+     {:fx [(when-not current-address
+             [:dispatch
+              [:wallet/switch-current-viewing-account address]])
+           [:dispatch
+            [:wallet.swap/select-asset-to-pay
+             {:token   token
+              :network (when (= count token-networks)
+                         (first token-networks))}]]]})))
+
 (rf/reg-event-fx :wallet.swap/select-asset-to-pay
  (fn [{:keys [db]} [{:keys [token network]}]]
    {:db (-> db
             (assoc-in [:wallet :ui :swap :asset-to-pay] token)
-            (assoc-in [:wallet :ui :swap :network] network))
+            (cond-> network (assoc-in [:wallet :ui :swap :network] network)))
```

Comment:
It's not great to mix these approaches, having a double threading macro makes things hard to understand.

Consider:
```clojure
(cond-> db
  :always (assoc-in [:wallet :ui :swap :asset-to-pay] token)
  network (assoc-in [:wallet :ui :swap :network] network))
```
instead